src/main/java/bio/terra/cli/businessobject/resource/BqDataset.java

```diff
@@ -183,4 +185,9 @@ public String getProjectId() {
   public String getDatasetId() {
     return datasetId;
   }
+
+  public int getTableCount() {
+    GoogleBqDatasets datasets = new GoogleBqDatasets(Context.requireUser().getPetSACredentials());
```

Comment:
could you add a factory method for this, similar to WSM service class `fromContextForPetSa()`?

Reply:
sure